src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -234,6 +246,19 @@ private static void LogErrors(ScriptHost host)
             }
         }
 
+        /// <summary>
+        /// Handler for any file changes under the root script path
+        /// </summary>
+        private void OnScriptFileChanged(object sender, FileSystemEventArgs e)
+        {
+            if (_restartDelayTokenSource != null && !_restartDelayTokenSource.IsCancellationRequested)
+            {
+                // if we're currently awaiting an error/restart delay
```

Comment:
Can you clarify this here? So any file changes under the script root path will halt restart? Is this just to avoid conflicts when actual script file changes are made? What about conditions that would not trigger a host restart?